### PR TITLE
rpki-client: update to version 8.2

### PR DESCRIPTION
**Pull request by artkiver**
* closes: https://trac.macports.org/ticket/66677
 * add variants for libressl, openssl (requires libretls)
 * add maintainer
 * added default_variants +libressl to appease the CI buildbots
 * other sanity improvements as suggested by neverpanic in closed PR 17298.

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 13.1 22C65 arm64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: net/rpki-client/Portfile [hunk 1]
```diff
@@ -26,10 +26,29 @@ homepage            https://www.rpki-client.org/index.html
...
+                    size   475141
 
-depends_lib-append  port:expat \
-                    port:libretls \
+depends_lib         port:expat
+
+variant libressl conflicts openssl description {use LibreSSL} {
+depends_lib-append  path:lib/libtls.dylib:libressl
+configure.args-append --without-libretls
+}
```
Comment (neverpanic) [inline net/rpki-client/Portfile:38]:
```suggestion
variant libressl conflicts openssl description {use LibreSSL} {
    depends_lib-append  path:lib/libtls.dylib:libressl
    configure.args-append --without-libretls
}
```

File: net/rpki-client/Portfile [hunk 2]
```diff
@@ -26,10 +26,29 @@ homepage            https://www.rpki-client.org/index.html
...
+
+variant openssl conflicts libressl description {use OpenSSL} {
+depends_lib-append  port:libretls \
                     port:openssl
+}
```
Comment (neverpanic) [inline net/rpki-client/Portfile:43]:
```suggestion
variant openssl conflicts libressl description {use OpenSSL} {
    depends_lib-append  port:libretls \
                        port:openssl
}
```

**Review by neverpanic — COMMENTED** [review]
Looks good, but I'd fix the indentation of the variants. The if-else-cascade for the default variants it indented with two spaces, but the modeline also says 4.

**Comment by artkiver** [comment]
My apologies for the delays, I was planning to add some livecheck parameters as well, going so far as to prototype amending the Portfile with this:

```
livecheck.type      regex
livecheck.url       https://ftp.openbsd.org/pub/OpenBSD/rpki-client/
livecheck.regex     "rpki-client-(\\d+\\.\\d+).tar.gz"
```

But then I noticed that https://ports.macports.org/port/rpki-client/details/ already seems to notice that the port is out of date despite no livecheck parameters being explicitly declared in the Portfile! So, hopefully your amended suggestions are sufficient for now.

Thank you!🙏

**Comment by artkiver** [comment]
(Whoops, I did not mean to close the PR, I just meant to comment. My apologies).

**Review by neverpanic — APPROVED** [review]

**Pull request merged**